app/models/course.rb

```diff
@@ -113,6 +113,23 @@ def root_folder?
     end
   end
 
+  def self_directed_learning_max_time
+    settings(:course).self_directed_learning_max_time
+  end
+
+  def self_directed_learning_max_time=(time)
```

Comment:
@weiqingtoh Actually do you have better names for these methods?  now I feel the method names are too long and hard to recall in other places..